### PR TITLE
Fix broken links after moving proposals to subdirs

### LINKED ISSUE #1109: Design proposals have lots of broken links to other design proposals
https://github.com/kubernetes/community/pull/1010 moved design proposals to SIG specific directories but didn't update the internal links from one design proposal to another. So, all proposals that refer to other proposals have broken links now which makes them harder to read.

/cc @grodrigues3 @bgrant0607

### PR DESCRIPTION
**Pull request by php-coder**
Fix broken links after kubernetes/community#1010
Because I fixed them semi-automatically it's possible that patch has some mistakes (but I hope there is none).

Fixes #1109

CC @grodrigues3 @bgrant0607 @simo5

### CODE REVIEW

**Comment by xiangpengzhao** [comment]
cc @zhangxiaoyu-zidif

**Comment by bgrant0607** [comment]
Thanks!
/lgtm

File: contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md [hunk 1]
```diff
@@ -239,8 +239,8 @@ The binding of the `system:bootstrappers` (or similar) group to the ability to s
 
 ## Revision history
 
- - Initial proposal ([@jbeda](https://github.com/jbeda)): [link](https://github.com/kubernetes/community/blob/cb9f198a0763e0a7540cdcc9db912a403ab1acab/contributors/design-proposals/bootstrap-discovery.md)
- - v1.6 updates ([@jbeda](https://github.com/jbeda)): [link](https://github.com/kubernetes/community/blob/d8ce9e91b0099795318bb06c13f00d9dad41ac26/contributors/design-proposals/bootstrap-discovery.md)
+ - Initial proposal ([@jbeda](https://github.com/jbeda)): [link](https://github.com/kubernetes/community/blob/cb9f198a0763e0a7540cdcc9db912a403ab1acab/contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md)
```
Comment (mindprince) [inline contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md]: This linked to a particular commit sha. So, the earlier URL was correct. The new one isn't.

File: contributors/design-proposals/scheduling/pod-preemption.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ _Author: @bsalamat_
...
+Running various types of workloads with different priorities is a common practice in medium and large clusters to achieve higher resource utilization. In such scenarios, the amount of workload can be larger than what the total resources of the cluster can handle. If so, the cluster chooses the most important workloads and runs them. The importance of workloads are specified by a combination of [priority](https://github.com/bsalamat/community/blob/564ebff843532faf5dcb06a7e50b0db5c5b501cf/contributors/design-proposals/scheduling/pod-priority-api.md), QoS, or other cluster-specific metrics. The potential to have more work than what cluster resources can handle is called "overcommitment". Overcommitment is very common in on-prem clusters where the number of nodes is fixed, but it can similarly happen in cloud as cloud customers may choose to run their clusters overcommitted/overloaded at times in order to save money. For example, a cloud customer may choose to run at most 100 nodes, knowing that all of their critical workloads fit on 100 nodes and if there is more work, they won't be critical and can wait until cluster load decreases.
```
Comment (mindprince) [inline contributors/design-proposals/scheduling/pod-preemption.md]: Same here.

File: contributors/design-proposals/scheduling/pod-preemption.md [hunk 3]
```diff
@@ -71,11 +71,11 @@ The race condition will still exist if we have multiple schedulers. More on this
...
+When scheduling a pending pod, scheduler tries to place the pod on a node that does not require preemption. If there is no such a node, scheduler may favor a node where the number and/or priority of victims (preempted pods) is smallest. After choosing the node, scheduler considers the lowest [priority](https://github.com/bsalamat/community/blob/564ebff843532faf5dcb06a7e50b0db5c5b501cf/contributors/design-proposals/scheduling/pod-priority-api.md) pods for preemption first. Scheduler starts from the lowest priority and considers enough pods that should be preempted to allow the pending pod to schedule. Scheduler only considers pods that have lower priority than the pending pod.
```
Comment (mindprince) [inline contributors/design-proposals/scheduling/pod-preemption.md]: Same here.

File: contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md [hunk 4]
```diff
@@ -239,8 +239,8 @@ The binding of the `system:bootstrappers` (or similar) group to the ability to s
...
+ - v1.6 updates ([@jbeda](https://github.com/jbeda)): [link](https://github.com/kubernetes/community/blob/d8ce9e91b0099795318bb06c13f00d9dad41ac26/contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md)
```
Comment (mindprince) [inline contributors/design-proposals/cluster-lifecycle/bootstrap-discovery.md]: Same here.

**Comment by php-coder** [comment]
@mindprince Good catch! Thank you! I've submitted https://github.com/kubernetes/community/pull/1118.

**Pull request merged**